### ISSUE TITLE
DNAME causes an incorrectly reported DoE error

### ISSUE THREAD

**Issue by vcunat** [dnsviz/dnsviz#152]
Example: https://dnsviz.net/d/_443._tcp.lg.ix.bzh/ZwkLgQ/dnssec/?rr=all&a=all&ds=all&doe=on&ta=.&tk=

- just a single name in the zone, so a self-looping NSEC denies existence of all other names
- DNAME in the root generates positive answers for those names
- DnsViz reports that as an error
  > NSEC proving non-existence of ix.bzh/CNAME: The following queries resulted in an answer response, even though the NSEC records indicate that the queried names don't exist: _443._tcp.lg.breizh-ix.net/TLSA See RFC 4035, Sec. 3.1.3.2.

**Comment by vcunat** [comment]
The error message itself is confusing, too.  The denied name is before following the generated CNAME (`_443._tcp.lg.ix.bzh`), but the error message mentions the target of the CNAME, which doesn't make sense to me.

**Comment by cdeccio** [comment]
> Example: https://dnsviz.net/d/_443._tcp.lg.ix.bzh/ZwkLgQ/dnssec/?rr=all&a=all&ds=all&doe=on&ta=.&tk=
> 
>     * just a single name in the zone, so a self-looping NSEC denies existence of all other names
> 
>     * DNAME in the root generates positive answers for those names
> 
>     * DnsViz reports that as an error
>       > NSEC proving non-existence of ix.bzh/CNAME: The following queries resulted in an answer response, even though the NSEC records indicate that the queried names don't exist: _443._tcp.lg.breizh-ix.net/TLSA See RFC 4035, Sec. 3.1.3.2.

Yeah, it looks like DNSViz is using the synthesized (i.e., from the DNAME) CNAME record and suggesting that it exists.

> The error message itself is confusing, too. The denied name is before following the generated CNAME (`_443._tcp.lg.ix.bzh`), but the error message mentions the target of the CNAME, which doesn't make sense to me.

Yes, this confusing.  I would need to look into why it is doing this.  I have some hunches, but it will take some time.

**Comment by cdeccio** [comment]
Oh, and thanks for reporting 😅

**Comment by cdeccio** [comment]
If only the DNS were easy... But that might put us out of work! 😂
